projects/bp-gallery/src/components/common/TagOverview.tsx

```diff
@@ -34,27 +34,33 @@ const TagOverview = ({
   archiveId,
 }: TagOverviewProps) => {
   const { t } = useTranslation();
+  const ref = useRef<any>();
 
   const basePath = archiveId
     ? '/archives/' + archiveId + '/show-more/' + type + '/'
     : '/show-more/' + type + '/';
 
-  const calculateMaxCategoriesPerRow = () => {
-    const tempRowLength = Math.max(1, Math.floor(Math.min(window.innerWidth - 64, 1200) / 300));
-    if (Math.min(window.innerWidth - 64, 1200) >= tempRowLength * 300 + (tempRowLength - 1) * 8) {
+  const calculateMaxCategoriesPerRow = useCallback((width: number) => {
+    const tempRowLength = Math.max(1, Math.floor(Math.min(width, 1200) / 260));
+    if (Math.min(width, 1200) >= tempRowLength * 260 + (tempRowLength - 1) * 8) {
       return tempRowLength;
     }
     return Math.max(1, tempRowLength - 1);
-  };
+  }, []);
 
   const [rowLength, setRowLength] = useState(() => {
-    const initialState = calculateMaxCategoriesPerRow();
+    const initialState = calculateMaxCategoriesPerRow((ref.current?.clientWidth ?? 0) as number);
```

Comment:
Since you don't do anything with the initialState, you can just immediately return the expression instead of saving it as a const.